### DISCUSSION TITLE
TypeError: 'MaskedQuantity' object with a scalar value is not iterable

### DISCUSSION THREAD

**Discussion by rfreire-400** [lightkurve/lightkurve#1338] [Q&A]
Hi,

I was trying to run the following code when I got this error message. It used to run just fine a couple of days ago but now it's giving me problems. Here's the relevant part of the code:

<img width="610" alt="Screenshot 2023-05-23 at 11 51 04 AM" src="https://github.com/lightkurve/lightkurve/assets/113457579/0326ead5-8c0d-4655-b16c-a617651bacae">

Normalize does not seem to be working like it used to. I also tried with flatten and it gave me the same issue.
The error message is very long, but at the end it says: `TypeError: 'MaskedQuantity' object with a scalar value is not iterable`

This used to work not even a week ago so I am quite perplexed that it doesn't anymore. Has anyone ran into a similar issue recently? Does anyone know how to fix this? Am I missing something?

Thank you for your time.

**Comment by Nschanche** [comment]
Hi, I just tried this and had no problem with the normalize function. Could you provide what version of Python/lightkurve/astropy/numpy you are using?

import sys; print("Python", sys.version)
import lightkurve as lk; print("lightkurve", lk.__version__)
import astropy; print("astropy", astropy.__version__)
import numpy as np; print("numpy", np.__version__)

**Reply by rfreire-400** [reply]
Hi,
Here are the versions that I'm using.
<img width="684" alt="Screenshot 2023-05-23 at 1 07 43 PM" src="https://github.com/lightkurve/lightkurve/assets/113457579/65240bec-db3d-41ba-8737-de0e7d2dc812">

Thanks

**Comment by rfreire-400** [comment]
It seems that, on my end at least, this error only occurs with very specific light curves for some reason. It will not run if the flux's type is a MaskedQuantity. After trying out some different light curves for mu Eridani, it seems that only light curves from SPOC or TESS-SPOC have their fluxes as MaskedQuantities:

<img width="926" alt="Screenshot 2023-05-24 at 2 00 14 PM" src="https://github.com/lightkurve/lightkurve/assets/113457579/68590f0a-cc7e-4c92-99aa-a34fce0ce53c">

I haven't tested this with other stars. Does this happen to anyone else as well?

**Comment by Nschanche** [comment]
Thanks for that great breakdown. I've turned this comment into an issue we can look into more.

**Comment by rfreire-400** [comment]
Upgrading astropy to version 5.3 solved the issue for me. With version 5.3, it's able to take in the flux as a MaskedQuantity and normalize it as intended. SPOC and TESS-SPOC measurements still give flux values in MaskedQuantity as opposed to Quantity with other sources. Unsure if that's by design or not.

**Comment by JohaMR** [comment]
I'm dealing with the same error, but I can't upgrade the astropy version because a part of the code failed when using the latest version. I would like to know if there is any solution to the TypeError: 'MaskedQuantity' object with a scalar value is not iterable problem.

**Reply by Nschanche** [reply]
Hi @JohaMR, could you please post the full error message you are receiving so I can see exactly what line in normalize() is causing the issue? Also, can you include the versions of the packages you are working with? You can also take a look at #1280, which addresses this issue in a different context and may help find a temporary fix. We are looking more deeply into MaskedQuantity handling now, but do not have a fix built in just yet.

**Comment by Enessyakut** [comment]
`lc_scsf = lk.search_lightcurve('KIC 11182716', author="Kepler", cadence='short').download_all().stitch()
ax = lc_scsf.remove_outliers().to_periodogram().plot()

for i in range(8):
  i+=1
  if i == 8:
    ax.annotate('', xy=(i*48.94, 1.1e-5), xytext=(i*48.94, 3e-5), arrowprops=dict(arrowstyle='->', color='r', linewidth=1.5))
  else:
    ax.annotate('', xy=(i*48.94, 1.1e-5), xytext=(i*48.94, 3e-5), arrowprops=dict(arrowstyle='->', color='r', linewidth=1.5, alpha=0.5))
`

When I write this code, I get the same 'Masked Quantity' object with a scalar value is not iterable, error. I also received this error in some different codes. I would be happy if you could help me.
